### PR TITLE
[RISCV] Support VLS for VCIX

### PR DESCRIPTION
**Pull request by 4vtomat**

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -8163,6 +8200,46 @@ SDValue RISCVTargetLowering::LowerINTRINSIC_W_CHAIN(SDValue Op,
...
+      }
+
+      MVT OpContainerVT =
+          getContainerForFixedLengthVector(V.getSimpleValueType());
+      Ops.push_back(convertToScalableVector(OpContainerVT, V, DAG, Subtarget));
+    }
+
+    SDLoc DL(Op);
+    MVT RetContainerVT = getContainerForFixedLengthVector(VT);
+    SDValue ScalableVector =
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:8237]: This should have two results, the RetContainerVT and MVT::Chain

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -8163,6 +8200,46 @@ SDValue RISCVTargetLowering::LowerINTRINSIC_W_CHAIN(SDValue Op,
...
+        DAG.getNode(ISD::INTRINSIC_W_CHAIN, DL, RetContainerVT, Ops);
+    SDValue FixedVector =
+        convertFromScalableVector(VT, ScalableVector, DAG, Subtarget);
+    return DAG.getMergeValues({FixedVector, Op.getOperand(0)}, DL);
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:8241]: `Op.getOperand(0)` should be `ScalableVector.getValue(1)` after making `ScalableVector` have two results.

**Review by topperc — CHANGES_REQUESTED** [review]

**Comment by 4vtomat** [comment]
ping

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**